api/src/test/java/org/pmiops/workbench/actionaudit/auditors/SumoLogicAuditorTest.java

```diff
@@ -0,0 +1,216 @@
+
+  @Captor private ArgumentCaptor<Collection<ActionAuditEvent>> eventsCaptor;
+
+  @Rule public final ExpectedException exception = ExpectedException.none();
+
+  @Before
+  public void setUp() {
+    dbUser = new DbUser();
+    dbUser.setUserId(USER_ID);
+    dbUser.setUsername(USER_EMAIL);
+    when(mockUserDao.findUserByUsername(USER_EMAIL)).thenReturn(dbUser);
+
+    dbWorkspace = new DbWorkspace();
+    dbWorkspace.setWorkspaceId(WORKSPACE_ID);
+    dbWorkspace.setWorkspaceNamespace(WORKSPACE_NAMESPACE);
+    dbWorkspace.setFirecloudName(WORKSPACE_FIRECLOUD_NAME);
+    when(mockWorkspaceService.getByNamespace(WORKSPACE_NAMESPACE)).thenReturn(dbWorkspace);
+    firecloudUserRoles.add(new UserRole().email(USER_EMAIL));
+    when(mockWorkspaceService.getFirecloudUserRoles(WORKSPACE_NAMESPACE, WORKSPACE_FIRECLOUD_NAME))
+        .thenReturn(firecloudUserRoles);
+  }
+
+  Set<String> extractValuesFromEvents(
+      Collection<ActionAuditEvent> events, Function<ActionAuditEvent, String> fn) {
+    return events.stream().map(fn).collect(Collectors.toSet());
+  }
+
+  @Test
+  public void testFireEgressEvent() {
+    sumoLogicAuditor.fireEgressEvent(
+        new EgressEvent()
+            .projectName(EGRESS_EVENT_PROJECT_NAME)
+            .vmName(EGRESS_EVENT_VM_NAME)
+            .timeWindowStart(0l)
+            .egressMib(12.3));
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    // Ensure all events have the expected set of constant fields.
+    assertThat(events.stream().map(event -> event.getAgentType()).collect(Collectors.toSet()))
+        .containsExactly(AgentType.USER);
+    assertThat(events.stream().map(event -> event.getActionType()).collect(Collectors.toSet()))
+        .containsExactly(ActionType.HIGH_EGRESS);
+    assertThat(events.stream().map(event -> event.getAgentId()).collect(Collectors.toSet()))
+        .containsExactly(USER_ID);
+    assertThat(events.stream().map(event -> event.getAgentEmailMaybe()).collect(Collectors.toSet()))
+        .containsExactly(USER_EMAIL);
+    assertThat(events.stream().map(event -> event.getTargetIdMaybe()).collect(Collectors.toSet()))
+        .containsExactly(WORKSPACE_ID);
+
+    // We should have distinct event rows with values from the egress event.
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventTargetProperty.EGRESS_MIB.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .collect(Collectors.toSet()))
+        .containsExactly("12.3");
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventTargetProperty.VM_NAME.getPropertyName())
```

Comment:
nit: for Java string comparisons, it's nice to put the known non-null string on the left so you can use `.equals()`

Reply:
Ah, I always forget about that. Done.